Cookbook/Proposals/SplitOctopus.md

```diff
@@ -0,0 +1,45 @@
+# Split Octopus codebase
+
+* Author(s): Giorgos Tsiapaliokas
+* Review Managers: David Rodrigues, Yasuhiro Inami, Olivier Halligon
+
+## Introduction
+
+From the early versions of the codebase Octopus was the place were features were added.
+Unfortunately Octopus hasn't adapted as the rest of the codebase has.
+In the iOS codebase we use frameworks in order to organize the code and Octopus isn't one.
+
+## Motivation
+
+While there is still the need for having a place for small features Octopus has become a massive container which contains both features and services.
+Also from a tech POV Octopus is just a folder which is being reused in different targets.
+I am sure we can do better than this :)
+
+## Proposed solution
+
+Split Octopus into two different frameworks.
+The proposed names are `BabylonFeatures` and `BabylonApp`.
+
+The first framework will be used as a container for small features and it should depend only on `BabylonCore`, `BabylonUI`, `BabylonDepedencies` and `BabylonSDK`.
+
+The second framework will contain anything else which was in Octopus.
+This framework can depend on any other framework.
+
+Converting Octopus into two frameworks
+- should improve the build time of the app
+- will reduce the time which is required for unit tests to be executed
+- will allow better code organization
+- engineers won't be limited in any way
+
+In the initial split `BabylonFeatures` will host all the features which satisfy the dependencies requirement.
+After the initial split it will be up to each squad to decide the appropriate place for their code.
```

Comment:
I'm not clear how you'd want to organise that split. I think it warrants more details

 - Which current subfolders of Octopus will go in which framework?
 - "BabylonFeatures will be use as a container for small features"
   - How do you define big vs small feature?
   - Would that mean that every feature and screen and renderer and builder would move into that framework? I guess not, so how would we decide which feature should go in one of our other Babylon* modules vs inside BabylonFeatures?
 - "BabylonApp will contain anything else" sounds like we plan to keep that framework (however we plan to name it) as a "dumping ground for everything we don't know where to put"?
    - Is that temporary (just the time to do a first pass extraction and triage the remaining code better later) or a plan for the long term?
    - In any case, not sure that the name `BabylonApp` is very representative here, could you give more examples of what you'd imagine would end up in that second framework?
 - How does that work with other flavors (Bupa, Telus, Ascension, ...)
    - question actually applies for both frameworks described here. Will BabylonFeatures contain only Babylon UK features or could it also end up containing Telus ones? Same for BabylonApp